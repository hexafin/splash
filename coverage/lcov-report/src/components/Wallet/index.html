<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Wallet</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Wallet
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">55.56% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>15/27</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">55.56% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>15/27</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="Wallet.js"><a href="Wallet.js.html">Wallet.js</a></td>
	<td data-value="56.52" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 56%;"></div><div class="cover-empty" style="width:44%;"></div></div></td>
	<td data-value="56.52" class="pct medium">56.52%</td>
	<td data-value="23" class="abs medium">13/23</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="8" class="abs medium">5/8</td>
	<td data-value="30" class="pct low">30%</td>
	<td data-value="10" class="abs low">3/10</td>
	<td data-value="56.52" class="pct medium">56.52%</td>
	<td data-value="23" class="abs medium">13/23</td>
	</tr>

<tr>
	<td class="file medium" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Apr 19 2019 16:14:10 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
