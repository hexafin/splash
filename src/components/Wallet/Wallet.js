import React, { Component } from "react";
import {
	View,
	Text,
	StyleSheet,
	Image,
	Share,
	ScrollView,
	Animated,
	Clipboard,
	TouchableWithoutFeedback,
	Dimensions
} from "react-native";
import { colors } from "../../lib/colors";
import { defaults, icons } from "../../lib/styles";
import FlatBackButton from "../universal/FlatBackButton"
import Button from "../universal/Button"
import ReactNativeHapticFeedback from 'react-native-haptic-feedback';

const SCREEN_WIDTH = Dimensions.get("window").width
const SCREEN_HEIGHT = Dimensions.get("window").height

class Wallet extends Component {

	constructor(props) {
		super(props)
		this.state = {
			isCopying: false
		}
		this.handleCopy = this.handleCopy.bind(this)
	}

	componentWillMount() {
        if (!this.props.loggedIn) {
            this.props.navigation.navigate("Landing");
        }
        this.animatedAddressCopy = new Animated.Value(1)
	}

	handleCopy() {
		this.setState(prevState => {
			return {
				...prevState,
				isCopying: true
			}
		})
		
		Clipboard.setString(this.props.address)
		ReactNativeHapticFeedback.trigger("impactLight", true)

		setTimeout(() => {
			this.setState(prevState => {
				return {
					...prevState,
					isCopying: false
				}
			})
		}, 1000)
	}

	render() {
		const { address, splashtag } = this.props

		const qrCode = "https://chart.googleapis.com/chart?chs=250x250&cht=qr&chl="+address

		return (
			<View style={styles.container}>
				<ScrollView>
					<View style={styles.header}>
						<Text style={styles.title}>Your splash wallet</Text>
					</View>
					<View style={styles.body}>
						<View style={styles.bodyTitleWrapper}>
							<Text style={styles.bodyTitle}>Receive bitcoin here</Text>
						</View>
						<TouchableWithoutFeedback
							onPress={this.handleCopy}
							onPressIn={() => {
								Animated.spring(this.animatedAddressCopy, {
									toValue: .8
								}).start()
							}}
							onPressOut={() => {
								Animated.spring(this.animatedAddressCopy, {
									toValue: 1,
									friction: 3,
									tension: 40
								}).start()
							}}>
							<Animated.View style={[styles.addressCopyWrapper, {
								transform: [{scale: this.animatedAddressCopy}]
							}]}>
								<Image source={{uri: qrCode}} style={styles.qr}/>
								<View style={styles.addressWrapper}>
									<Text style={styles.addressText}>{address}</Text>
								</View>
								<Text style={styles.addressCopyText}>
									{this.state.isCopying && "Copied!"}
									{!this.state.isCopying && "Tap to copy"}
								</Text>
							</Animated.View>
						</TouchableWithoutFeedback>
						<Button small title="Share" onPress={() => {
							Share.share({
								title: `@${splashtag} bitcoin wallet`,
								message: address
							})
						}} style={{
							marginTop: 30,
							width: 200
						}}/>
					</View>
				</ScrollView>
			</View>
		);
	}
}

const styles = StyleSheet.create({
	container: {
		...defaults.container,
		backgroundColor: "transparent"
	},
	header: {
		height: 150,
		justifyContent: "center",
		alignItems: "center",
		marginBottom: 40,
		paddingTop: 60
	},
	title: {
		color: colors.white,
		fontSize: 24,
		fontWeight: "700"
	},
	body: {
		flex: 1,
		padding: 20,
		flexDirection: "column",
		alignItems: "center",
		justifyContent: "flex-start",
	},
	bodyTitleWrapper: {
		width: "100%",
		flexDirection: "row",
		justifyContent: "flex-start",
		marginBottom: 10
	},
	bodyTitle: {
		color: colors.primaryDarkText,
		fontSize: 20,
		fontWeight: "700",
		textAlign: "left"
	},
	qr: {
		width: 240,
		height: 240,
		padding: 40
	},
	addressCopyWrapper: {
		flexDirection: "column",
		alignItems: "center",
	},
	addressWrapper: {
		backgroundColor: colors.primaryLight,
		paddingLeft: 10,
		paddingRight: 10,
		paddingTop: 6,
		paddingBottom: 6,
		borderRadius: 3,
		marginBottom: 10
	},
	addressText: {
		color: colors.primary,
		fontSize: 14,
		fontWeight: "700"
	},
	addressCopyText: {
		fontSize: 14,
		fontWeight: "600",
		color: colors.lightGray
	}
});

export default Wallet;