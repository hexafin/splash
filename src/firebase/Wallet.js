// creates hexa wallet (different from bitcoin wallet)
// returns wallet entity

NewWallet = function(creator_id, type, picture_url, hex, members=[], transactions=[]) {

    // TODO

};

export default NewWallet;