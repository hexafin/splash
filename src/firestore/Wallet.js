
function NewPersonalWallet(hex, description) {

    // create hexa wallet

    // create bitcoin wallet
    // store private key locally in private_keys object {wallet_id:private_key,...}

    // update hexa wallet with bitcoin wallet's public key

}

function NewGroupWallet(hex, description, members, ownership, signing_members) {

    // create hexa wallet

    // create group bitcoin wallet with signing structure
    // store private key locally in private_keys object {wallet_id:private_key,...}

    // update hexa wallet with bitcoin wallet's public key

}

export { NewPersonalWallet, NewGroupWallet };