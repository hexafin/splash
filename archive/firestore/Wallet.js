/*

Wallet.js defines firestore operations with hexa wallets

 */


import firebase from 'react-native-firebase';
let firestore = firebase.firestore();
var random = require('react-native-randombytes').randomBytes;
var bitcoin = require('bitcoinjs-lib');
var axios = require('axios');

import { NewInternalChat } from "./Chat";

function NewPersonalWallet(personRef, description) {
    return new Promise((resolve, reject) => {
        // create hexa wallet
        personRef.get().then(person => {

            var hex = GenerateHex(person.data()["first_name"].toLowerCase(), person.data()["last_name"].toLowerCase());

            var walletName = person.data()["first_name"] + " " + person.data()["last_name"];

            // create bitcoin wallet
            var bitcoinWallet = NewBitcoinWallet();

            // TODO: store key locally in wallet_id object {bitcoin:WIF,ethereum:ethWiF,...}
            var wif = bitcoinWallet.wif;

            firestore.collection("wallets").add({
                type: "personal",
                address_bitcoin: bitcoinWallet.address,
                hex: hex,
                name: walletName,
                description: description,
                picture_url: person.data()["picture_url"],
                person: personRef
            }).then(walletRef => {
                resolve(walletRef);
            });

        });
    });
}

// TODO: NewGroupWallet
function NewGroupWallet(hex, description, memberRefs, ownership, signingMemberRefs) {

    // create hexa wallet
    var redeemScript = bitcoin.script.multisig.output.encode(2, pubKeys); // 2 of 3
    var scriptPubKey = bitcoin.script.scriptHash.output.encode(bitcoin.crypto.hash160(redeemScript));
    var address = bitcoin.address.fromOutputScript(scriptPubKey);



    // create internal chat from wallet ref
    NewInternalChat()


    // TODO: Group Wallet ownership
    // var ownership = {};
    // ownership[personRef.id] = 1.0;

    // create group bitcoin wallet with signing structure
    // store private key locally in private_keys object {wallet_id:private_key,...}

    // update hexa wallet with bitcoin wallet's public key

}

// GenerateHex finds a unique hex from a person's first and last name
function GenerateHex(firstName, lastName, num=0) {

    // initialize hex
    var hex = firstName + "-" + lastName;
    if (num > 0) {
        hex = hex + num.toString();
    }

    if (HexExists(hex)) {
        return GenerateHex(firstName, lastName, num=num+1);
    }
    else {
        return hex;
    }

}

function HexExists(hex) {
    // check if hex already exists
    firestore.collection("wallets").where("hex", "=", hex).get().then(checkHex => {

        if (checkHex.empty) {
            return false;
        }
        else {
            return true;
        }

    });
}

function NewBitcoinWallet() {
    var keyPair = bitcoin.ECPair.makeRandom({
        rng: random
    });
    return {
        keyPair: keyPair,
        wif: keyPair.toWIF(),
        address: keyPair.getAddress()
    };
}

export { NewPersonalWallet, NewGroupWallet, GenerateHex, HexExists };
