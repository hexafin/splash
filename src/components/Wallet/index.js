import Wallet from "./Wallet"
import { connect } from "react-redux"
import { bindActionCreators } from "redux"

const mapStateToProps = state => {
	return {
		address: state.crypto.wallets.BTC.address,
		loggedIn: state.user.loggedIn
	}
}

const mapDispatchToProps = dispatch => {
	return bindActionCreators({}, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Wallet)
