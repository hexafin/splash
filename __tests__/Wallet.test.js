import "react-native";
import React from "react";
import Wallet from "../src/components/Wallet/Wallet";

it("Wallet renders correctly", () => {
	const snap = shallow(
		<Wallet
			navigation={{
				state: {
					params: {}
				},
				navigate: () => {}
			}}
		/>
	);
	expect(snap).toMatchSnapshot();
});
