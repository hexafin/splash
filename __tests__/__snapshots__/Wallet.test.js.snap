// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renders correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
      "paddingBottom": 20,
      "paddingTop": 20,
    }
  }
>
  <ScrollView>
    <Component
      style={
        Object {
          "alignItems": "center",
          "height": 150,
          "justifyContent": "center",
          "marginBottom": 40,
          "paddingTop": 60,
        }
      }
    />
    <Component
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "padding": 20,
        }
      }
    >
      <Component
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginBottom": 10,
            "width": "100%",
          }
        }
      >
        <Component
          style={
            Object {
              "color": "#292342",
              "fontSize": 20,
              "fontWeight": "700",
              "textAlign": "left",
            }
          }
        >
          Receive 
           here
        </Component>
      </Component>
      <TouchableWithoutFeedback
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
      >
        <AnimatedComponent
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "column",
              },
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              },
            ]
          }
        >
          <Component
            source={
              Object {
                "uri": "https://chart.googleapis.com/chart?chs=250x250&cht=qr&chl=undefined",
              }
            }
            style={
              Object {
                "height": 240,
                "padding": 40,
                "width": 240,
              }
            }
          />
          <Component
            style={
              Object {
                "backgroundColor": "#EFEFFD",
                "borderRadius": 3,
                "marginBottom": 10,
                "paddingBottom": 6,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 6,
              }
            }
          >
            <Component
              style={
                Array [
                  Object {
                    "color": "#4547E0",
                    "fontWeight": "700",
                  },
                  Object {
                    "fontSize": 11,
                  },
                ]
              }
            />
          </Component>
          <Component
            style={
              Object {
                "color": "#ADADAD",
                "fontSize": 14,
                "fontWeight": "600",
              }
            }
          >
            Tap to copy
          </Component>
        </AnimatedComponent>
      </TouchableWithoutFeedback>
      <Button
        onPress={[Function]}
        small={true}
        style={
          Object {
            "marginTop": 30,
            "width": 200,
          }
        }
        title="Share"
      />
    </Component>
  </ScrollView>
</Component>
`;
